Schema/__init__.py

```diff
@@ -246,34 +245,29 @@ def update_with_reference_geometry(imagesets, reference_geometry_list):
 
 
 def load_reference_geometries(geometry_file_list):
-    from dxtbx.serialize import load
-
-    reference_components = []
-    for file in geometry_file_list:
-        try:
-            experiments = load.experiment_list(file, check_format=False)
-            assert len(experiments.detectors()) == 1
-            assert len(experiments.beams()) == 1
-            reference_detector = experiments.detectors()[0]
-            reference_beam = experiments.beams()[0]
-        except Exception:
-            experiments = load.experiment_list(file)
-            imageset = experiments.imagesets()[0]
-            reference_detector = imageset.get_detector()
-            reference_beam = imageset.get_beam()
-        reference_components.append(
-            {"detector": reference_detector, "beam": reference_beam, "file": file}
-        )
+    logger.debug("Collecting reference instrument models.")
+    ref_components = [
+        (expt.detector, expt.beam, f, i)
+        for f in geometry_file_list
+        for i, expt in enumerate(ExperimentList.from_file(f, check_format=False))
+    ]
+
+    logger.debug("Removing duplicate reference geometries.")
+    unique = set()
+    for a, b in filter(unique.isdisjoint, itertools.combinations(ref_components, 2)):
+        if compare_geometries(a[0], b[0]):
+            # Note that expt_index is the index of the experiment in the expt list file,
+            # as per dials.show, rather than the UID string of the experiment.
+            logger.debug(f"Experiment {b[3]} of {b[2]} is a duplicate.")
+        else:
+            unique.update((a, b))
```

Comment:
Suggestion:
```suggestion
            unique.add(a)
            unique.add(b)
```
does the same, but feels more obvious to me. I can't remember seeing `.update()` on a `set` before

Reply:
Sure thing.